dev-tools.asciidoc

```diff
@@ -367,13 +367,30 @@ Website link; https://openzeppelin.org/
+to develop and manage smart contract applications securely. 
 
-Github link; https://github.com/zeppelinos
+Unlike OpenZeppelin's code, which needs to be re-deployed with each application every time, zeppelin_os's code lives on-chain. Applications that need a given functionality, say, an ERC20 token, not only don't have to re-design and re-audit its implementation (something that OpenZeppelin solved) but don't even need to deploy it. With zeppelin_os, an application interacts with the token's on-chain implementation directly, much in the same way a desktop application interacts with the components of its underlaying OS.
+
+Applications that make use of OpenZeppelin avoid having to "reinvent the wheel" by reusing the library's curated and peer reviewed contracts. However, every single time an application uses an ERC20 token implementation, the same ERC20 bytecode is deployed into the blockchain, over and over again. Such bytecode per-se, exists duplicated in the network countless times. Now, applications that make use of zeppelin_os avoid this unnecessary duplication. Instead of deploying their own ERC20 implementation, they link to a contract that defines the latest ERC20 implementation accepted by the community. This single, central implementation is deployed in the blockchain only once, much like Solidity's libraries, but are considerably more sophisticated.
+
+Unlike Solidity's libraries, the implementations offered by zeppelin_os can be treated like regular contracts and, are upgradeable. If ever a vulnerability is found in one of the OS's official implementations, it can simply be swapped with an upgraded one. In the case of the ERC20 token, an upgrade to its implementation would instantly ripple to all the applications that use it. The OS not only provides upgradeability for all of its implementations, but also for the user's own contracts, and even for it's own codebase! Developers decide when and how an application should implement an upgrade, and even which "flavor" of an implementation to adhere to. 
+
+At the core of zeppelin_os sits a very clever contract known as a "proxy". A proxy is a contract that is capable of wrapping any other contract, exposing its interface without having to manually implement setters and getters for it, and can upgrade it without losing state. In Solidity terms, it can be seen as a normal contract whose business logic is contained within a library, which can be swapped by a new library at any time without losing its state. The way in which the proxy links to its implementation is completely automated and encapsulated for the developer. Practically any contract can be made upgradeable with little to no change in its code. More about zeppelin_os's proxy mechanism can be found in Zeppelin's blog: https://blog.zeppelinos.org/[upgradeability-using-unstructured-storage], and an example of how to use it can be found here: https://github.com/zeppelinos/zos-lib/tree/master/examples/single.
```

Comment:
I would remove "without losing its state" from ".. library at any time without losing its state" since it repeats from previous phrase.